src/avi/buffer/operate.clj

```diff
@@ -59,18 +59,22 @@
     [[(dec si) (count (get lines (dec si)))]
      [ei (count (get lines ei))]]))
 
+(defn fix-last-explicit-j
+  [[start [ei ej]] {:keys [last-explicit-j]}]
+  [start [ei (or ej last-explicit-j)]])
+
 (defn resolve-range
   [{:keys [lines point last-explicit-j] :as buffer} {:keys [span] :as operation}]
-  (when-let [[i j :as pos] (resolve/resolve-motion buffer operation)]
-    (let [j (or j last-explicit-j)]
-      (-> [point [i j]]
-        sort
-        (adjust-for-span lines span)))))
+  (when-let [range (resolve/resolve-motion buffer operation)]
```

Comment:
Been too busy lately... sorry for the delay!

Not a blocker, but you could consolidate the `when-let` and the `->` using `some->`.